In [1]:
import pandas as pd
import os
import json
import requests

In [18]:
p = 'A:/DataLake/Phase4/'
ps_finan = f'{p}Financial/'
ps_y = f'{ps_finan}Year/'
ps_q = f'{ps_finan}Quarter/'

ps_y_BS = f'{ps_y}BalanceSheet/'
ps_y_IS = f'{ps_y}IncomeStatement/'
ps_y_CF = f'{ps_y}CashFlow/'

ps_q_BS = f'{ps_q}BalanceSheet/'
ps_q_IS = f'{ps_q}IncomeStatement/'
ps_q_CF = f'{ps_q}CashFlow/'

p_Dividend = f'{p}Dividend/'
p_Volume = f'{p}Volume/'

p_DividendCash = f'{p}Dividend/DividendCash/'
p_DividendShares = f'{p}Dividend/DividendShares/'
p_TreasuryShares = f'{p}Volume/TreasuryShares/'
p_VolumeAdditionalEvents = f'{p}Volume/VolumeAdditionalEvents/'
p_VolumeNow = f'{p}Volume/VolumeNow.csv/'
p_DataUpDownExchange = f'{p}DataUpDownExchange/'
p_DataDownExchange = f'{p}DataUpDownExchange/DataDownExchange/'

all_com = pd.read_excel('base/Phase1/List_Com_Phase4.xlsx')['Symbol']

In [10]:
def list_col(all_com, path, list_col, list_false):
    list_ = []
    for i in all_com:
        # if i not in list_com_nothing:
            # print(i, end='  ')
            try:
                data = pd.read_csv(path+i+'.csv')
                if len(data.index) == 0 and len(data.columns) == 1:
                    continue
                if len(data.index) not in list_col:
                    # print(len(data), end= '  ')
                    list_.append(i)
                else:
                    # print('hi')
                    if ('Year' in path) and ('/' in str(list(data.columns))):
                        list_.append(i)
                        # print('hihi')
                    elif ('Quarter' in path) and ('/' not in str(list(data.columns))):
                        # print(i)
                        list_.append(i)
                    elif len(data.columns) <=3:
                        list_.append(i)
            except:
                # print(i)
                list_.append(i)
                pass
    print(len(list_))
    list_false[str(path)] = list_
    return list_false

In [25]:
# [40,56,96,0], , [30]
list_false = {}
list_col(all_com, ps_y_BS, [129], list_false)
list_col(all_com, ps_q_BS, [129], list_false)
list_col(all_com, ps_y_IS, [30], list_false)
list_col(all_com, ps_q_IS, [30], list_false)
list_col(all_com, ps_y_CF, [40,56,96], list_false)
list_col(all_com, ps_q_CF, [40,56,96], list_false)
with open('false.json', 'w') as f:
    json.dump(list_false, f)

0
73
0
73
0
73


In [23]:
list_folder= [p_DividendCash, p_DividendShares, p_TreasuryShares,p_VolumeAdditionalEvents,p_VolumeNow, p_DataDownExchange]

list_false_2 = {}
all_com = pd.read_excel('base/Phase1/List_Com_Phase4.xlsx')['Symbol']

for folder in list_folder:
    print(folder)
    list_ = []
    for i in all_com:
        try:
            data = pd.read_csv(folder+i+'.csv')
            if len(data.index) == 0 and len(data.columns) == 1:
                list_.append(i)
        except:
            list_.append(i)
            pass
    print(len(list_))
    list_false_2[str(folder)] = list_
with open('false2.json', 'w') as f:
    json.dump(list_false_2, f)

A:/DataLake/Phase4/Dividend/DividendCash/
120
A:/DataLake/Phase4/Dividend/DividendShares/
340
A:/DataLake/Phase4/Volume/TreasuryShares/
386
A:/DataLake/Phase4/Volume/VolumeAdditionalEvents/
272
A:/DataLake/Phase4/Volume/VolumeNow.csv/
48
A:/DataLake/Phase4/DataUpDownExchange/DataDownExchange/
390


A:/DataLake/Phase4/Dividend/DividendCash/
131
A:/DataLake/Phase4/Dividend/DividendShares/
343
A:/DataLake/Phase4/Volume/TreasuryShares/
387
A:/DataLake/Phase4/Volume/VolumeAdditionalEvents/
277
A:/DataLake/Phase4/Volume/VolumeNow.csv/
48
A:/DataLake/Phase4/DataUpDownExchange/DataDownExchange/
392

A:/DataLake/Phase4/Dividend/DividendCash/
120
A:/DataLake/Phase4/Dividend/DividendShares/
340
A:/DataLake/Phase4/Volume/TreasuryShares/
386
A:/DataLake/Phase4/Volume/VolumeAdditionalEvents/
272
A:/DataLake/Phase4/Volume/VolumeNow.csv/
48
A:/DataLake/Phase4/DataUpDownExchange/DataDownExchange/
390